examples/expressions/10_expressions_intro_new.py

```diff
@@ -0,0 +1,201 @@
+unseen_data = fetch_employee_salaries()
+unseen_data = unseen_data.employee_salaries[8000:]
+
+# %%
+# Then, we can use the loaded model to make predictions on the unseen data by
+# passing the environment as dictionary.
+
+predicted_values = loaded_model.predict({"data": unseen_data})
+predicted_values
+
+# %%
+# We can also evaluate the model's performance using the `score` method, which
+# uses the scikit-learn scoring function used by the predictor:
+loaded_model.score({"data": unseen_data})
+
+# %%
+# So far, we have seen how to build a simple machine learning pipeline using skrub's
+# DataOps. However, a major part of optimizing a machine learning model is
+# hyperparameter tuning. Skrub's DataOps allow us to easily add hyperparameters
+# right where we define the model, by using one of the `choose_*` functions.
+# Here, we will use `choose_float` to add a learning rate hyperparameter
+# to the `HistGradientBoostingRegressor`.
+
+hgb = HistGradientBoostingRegressor(
+    learning_rate=skrub.choose_float(0.01, 0.9, log=True, name="learning_rate")
+)
+predictor = X_vec.skb.apply(hgb, y=y)
+predictor
+
+# %%
+# Since now we have can tune the learning rate, we can use the `get_randomized_search`
+# method to perform hyperparameter tuning. This method will automatically
+# create a randomized search over the hyperparameters defined in the DataOps plan.
+
+search = predictor.skb.get_randomized_search(
+    n_iter=4,
+    n_jobs=4,
+    random_state=0,
+    scoring="neg_mean_absolute_error",
+    fitted=True,
+)
+
+# %%
+# We can print the results of the search, along with various hyperparameter
+# configurations. In this case, we only have one hyperparameter: the learning rate.
+
+search.results_
+# %%
+# Another powerful tool to find the best hyperparameters provided by the skrub
+# DataOps is the parallel coordinates plot. This plot allows to visualize each
+# hyperparameter configuration as a line that connects the hyperparameter values
+# to the corresponding score.
+
+search.plot_results()
+# %%
+# Finally, we can retrieve the best learner from the search results, and save it
+# to disk. This learner will contain the best hyperparameter configuration
+# found during the search, and can be used to make predictions on new data.
+
+best_learner = search.best_pipeline_
+saved_model = pickle.dumps(best_learner)
+
+# %%
+# In this example, we have briefly introduced the skrub DataOps, and how they can
+# be used to build powerful machine learning pipelines. We have seen how to preprocess
+# data, train a model, and tune hyperparameters using skrub's DataOps. We have also
+# shown how to save and load the trained model, and how to make predictions on new
+# data using the trained model.
+#
+# However, skrub DataOps are significantly more powerful than what we have shown here.
+# The following examples cover specific use cases, such as multi-table machine
+# learning, complex hyperparameter tuning, and sampling, in more detail.
```

Comment:
Can you add links here, potentially to https://skrub-data.org/dev/auto_examples/index.html#skrub-expressions (using the right rst anchor, not the directly URL), to have something that requires less subsequent modifications.